packages/cadl-vs/src/VSExtension.cs

```diff
@@ -108,21 +121,24 @@ public async Task OnLoadedAsync()
         }
 
 #if VS2019
-    public Task OnServerInitializeFailedAsync(Exception e) {
-      Debug.Fail("Failed to initialize cadl-server:\r\n\r\n" + e);
-      return Task.CompletedTask;
-    }
+        public Task OnServerInitializeFailedAsync(Exception e)
+        {
+            var message = e is CadlUserErrorException ? e.Message : e.ToString();
+            Debug.Fail("Failed to initialize cadl-server:\r\n\r\n" + message);
+            return Task.CompletedTask;
+        }
 #endif
 
 #if VS2022
     public Task<InitializationFailureContext?> OnServerInitializeFailedAsync(ILanguageClientInitializationInfo initializationState) {
       var exception = initializationState.InitializationException;
+      var message = exception is CadlUserErrorException 
+        ? exception.Message 
+        : $"File issue at https://github.com/microsoft/cadl\r\n\r\n{exception?.ToString()}";
```

Comment:
nit: ToString is called automatically during formatting.

```suggestion
        : $"File issue at https://github.com/microsoft/cadl\r\n\r\n{exception}";
```